plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlVectorType.java

```diff
@@ -0,0 +1,226 @@
+            assertThat(query("SELECT v FROM " + table.getName()))
+                    .matches("VALUES ARRAY[REAL '1.0'], ARRAY[REAL '1.0', REAL '2.0'], ARRAY[REAL '1.0', REAL '2.0', REAL '3.0']");
+
+            onRemoteDatabase().execute("SELECT v <-> '[4,5,6]' FROM " + table.getName() + " WHERE id = 3");
+
+            assertThatThrownBy(() -> onRemoteDatabase().execute("SELECT v <-> '[4,5,6]' FROM " + table.getName()))
+                    .hasMessageContaining("different vector dimensions");
+            assertThatThrownBy(() -> onRemoteDatabase().execute("INSERT INTO " + table.getName() + " VALUES (4, '[]')"))
+                    .hasMessageContaining("vector must have at least 1 dimension");
+        }
+    }
+
+    @Test
+    void testVectorMaxDimension()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase(), "test_vector_max", "(v vector(16000))")) {
+            String postgresValue = IntStream.rangeClosed(1, 16000).mapToObj(String::valueOf).collect(joining(","));
+            onRemoteDatabase().execute("INSERT INTO " + table.getName() + " VALUES ('[" + postgresValue + "]')");
+
+            String trinoValue = IntStream.rangeClosed(1, 16000).mapToObj("REAL '%s'"::formatted).collect(joining(","));
+            assertThat(query("SELECT v FROM " + table.getName()))
+                    .matches("VALUES ARRAY[" + trinoValue + "]");
+        }
+    }
+
+    @Test
+    void testVectorUnsupportedDimension()
+    {
+        assertThatThrownBy(() -> onRemoteDatabase().execute("CREATE TABLE test_vector_unsupported (v vector(-1))"))
+                .hasMessageContaining("dimensions for type vector must be at least 1");
+        assertThatThrownBy(() -> onRemoteDatabase().execute("CREATE TABLE test_vector_unsupported (v vector(0))"))
+                .hasMessageContaining("dimensions for type vector must be at least 1");
+        assertThatThrownBy(() -> onRemoteDatabase().execute("CREATE TABLE test_vector_unsupported (v vector(16001))"))
+                .hasMessageContaining("dimensions for type vector cannot exceed 16000");
+    }
+
+    @Test
+    void testEuclideanDistanceCompatibility()
+    {
+        // Verify compatibility between Trino euclidean_distance function and pgvector <-> operator
+        try (TestTable table = new TestTable(onRemoteDatabase(), "test_vector", "(v vector(3))");
+                TestView view = new TestView(onRemoteDatabase(), "test_euclidean_distance", "SELECT v <-> '[4,5,6]' FROM " + table.getName())) {
+            onRemoteDatabase().execute("INSERT INTO " + table.getName() + " VALUES ('[1,2,3]')");
+
+            assertThat(query("SELECT euclidean_distance(v, ARRAY[4,5,6]) FROM " + table.getName()))
+                    .matches("SELECT * FROM tpch." + view.getName());
+
+            assertThatThrownBy(() -> onRemoteDatabase().execute("SELECT v <-> '[NaN]' FROM " + table.getName()))
```

Comment:
What if we apply `euclidean_distance` on `NaN` ? Is it restricted ?

Reply:
It's not restricted: 
https://github.com/trinodb/trino/blob/17f94ca9df082de2792f42871f5e72b8077acba7/core/trino-main/src/test/java/io/trino/operator/scalar/TestArrayVectorFunctions.java#L112-L117